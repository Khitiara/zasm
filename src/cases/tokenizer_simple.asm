; Config end
page_table equ 0x03
mapping_2m equ 0x83

  movd eax, 0x1000
  movw [eax + 0x000], \
    page_table | 0x2000 ; Write page table root
  movw [0x1FF8], page_table | 0x2000 ; Upper half is same
  movd cr3, eax

  movw bx, 0x2018
  movw ax, 0x6003
morel1map:
  movw [bx], ax
  subb ah, 0x10
  subb bl, 8
  jcc morel1map

  movw [0x2FF0], page_table | 0x3000 ; -2G
  movw [0x2FF8], page_table | 0x4000 ; -1G

  ; Indentity map bottom 2G
  movw di, 0x3000
  xorw ax, ax
moremappings:
  movb [di], mapping_2m ; Third level
  movw [di + 2], ax
  addw di, 8
  addw ax, 32
  jcc moremappings
