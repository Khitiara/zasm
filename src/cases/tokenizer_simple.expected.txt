newline "; Config end"
identifier "page_table"
keyword_equ "equ"
number_literal "0x03"
newline ""
identifier "mapping_2m"
keyword_equ "equ"
number_literal "0x83"
newline ""
newline ""
identifier "movd"
identifier "eax"
comma ","
number_literal "0x1000"
newline ""
identifier "movw"
l_square_bracket "["
identifier "eax"
plus "+"
number_literal "0x000"
r_square_bracket "]"
comma ","
identifier "page_table"
pipe "|"
number_literal "0x2000"
newline "; Write page table root"
identifier "movw"
l_square_bracket "["
number_literal "0x1FF8"
r_square_bracket "]"
comma ","
identifier "page_table"
pipe "|"
number_literal "0x2000"
newline "; Upper half is same"
identifier "movd"
identifier "cr3"
comma ","
identifier "eax"
newline ""
newline ""
identifier "movw"
identifier "bx"
comma ","
number_literal "0x2018"
newline ""
identifier "movw"
identifier "ax"
comma ","
number_literal "0x6003"
newline ""
identifier "morel1map"
colon ":"
newline ""
identifier "movw"
l_square_bracket "["
identifier "bx"
r_square_bracket "]"
comma ","
identifier "ax"
newline ""
identifier "subb"
identifier "ah"
comma ","
number_literal "0x10"
newline ""
identifier "subb"
identifier "bl"
comma ","
number_literal "8"
newline ""
identifier "jcc"
identifier "morel1map"
newline ""
newline ""
identifier "movw"
l_square_bracket "["
number_literal "0x2FF0"
r_square_bracket "]"
comma ","
identifier "page_table"
pipe "|"
number_literal "0x3000"
newline "; -2G"
identifier "movw"
l_square_bracket "["
number_literal "0x2FF8"
r_square_bracket "]"
comma ","
identifier "page_table"
pipe "|"
number_literal "0x4000"
newline "; -1G"
newline ""
newline "; Indentity map bottom 2G"
identifier "movw"
identifier "di"
comma ","
number_literal "0x3000"
newline ""
identifier "xorw"
identifier "ax"
comma ","
identifier "ax"
newline ""
identifier "moremappings"
colon ":"
newline ""
identifier "movb"
l_square_bracket "["
identifier "di"
r_square_bracket "]"
comma ","
identifier "mapping_2m"
newline "; Third level"
identifier "movw"
l_square_bracket "["
identifier "di"
plus "+"
number_literal "2"
r_square_bracket "]"
comma ","
identifier "ax"
newline ""
identifier "addw"
identifier "di"
comma ","
number_literal "8"
newline ""
identifier "addw"
identifier "ax"
comma ","
number_literal "32"
newline ""
identifier "jcc"
identifier "moremappings"
newline ""
eof ""
